### PR TITLE
Match weight-sharing option of GATConv in DGL

### LINKED ISSUE #4060: [cugraph-dgl] improve alignment of `GATConv` with DGL's native version
As summarized in [here](https://nvbugspro.nvidia.com/bug/4414747), our GATConv behaves differently than DGL's counterpart when handling bipartite feature inputs. We should align with DGL's convention.

### PR DESCRIPTION
**Pull request by tingyu66**
Closes #4060

### CODE REVIEW

**Review by alexbarghi-nv — APPROVED** [review]
I'm not too familiar with the code in question but this looks fine to me.  I would wait until Vibhu reviews before merging.

File: python/cugraph-dgl/tests/nn/test_transformerconv.py
Comment (VibhuJawa) [inline python/cugraph-dgl/tests/nn/test_transformerconv.py:45]: Nitpick:  Any particular reason you hard-coded `torch.device("cuda:0")` ?

Maybe we do below :

```
device = torch.device("cuda")
```

Which it think is equivalent too: 
```
device_index = torch.cuda.current_device()
current_device = torch.cuda.device(device_index)
```

**Review by VibhuJawa — APPROVED** [review]
Looks good apart from a non blocking nitpick.

File: python/cugraph-dgl/tests/nn/test_transformerconv.py
Comment (tingyu66) [reply python/cugraph-dgl/tests/nn/test_transformerconv.py]: :+1:  Changed to using current device.

**Review by VibhuJawa — APPROVED** [review]
LGTM

**Pull request merged**
